packages/lab/src/carousel/CarouselSlide.tsx

```diff
@@ -1,76 +1,134 @@
-import { makePrefixer } from "@salt-ds/core";
+import { makePrefixer, useForkRef } from "@salt-ds/core";
+import { useComponentCssInjection } from "@salt-ds/styles";
+import { useWindow } from "@salt-ds/window";
+import { clsx } from "clsx";
 import {
-  type ElementType,
   type HTMLAttributes,
-  type ReactElement,
+  type ReactNode,
   forwardRef,
+  useEffect,
   useRef,
+  useState,
 } from "react";
-
-import { clsx } from "clsx";
+import { useCarousel } from "./CarouselContext";
+import carouselSlideCss from "./CarouselSlide.css";
 
 export interface CarouselSlideProps extends HTMLAttributes<HTMLDivElement> {
-  ButtonBar?: ElementType;
-  Media: ReactElement;
-  description?: string;
-  title?: string;
-  contentAlignment?: "center" | "left" | "right";
+  /**
+   * Actions to be displayed in the content footer.
+   */
+  actions?: ReactNode;
+  /**
+   * The media to be displayed inside the slide.
+   */
+  media?: ReactNode;
+  /**
+   * The appearance of the slide. Options are 'bordered', and 'transparent'.
+   * 'transparent' is the default value.
+   * */
+  appearance?: "bordered" | "transparent";
+  /**
+   * Header text
+   */
+  header?: ReactNode;
 }
 
 const withBaseName = makePrefixer("saltCarouselSlide");
 
 export const CarouselSlide = forwardRef<HTMLDivElement, CarouselSlideProps>(
   function CarouselSlide(
-    { ButtonBar, Media, description, title, contentAlignment },
-    ref,
+    {
+      actions,
+      appearance,
+      media,
+      header,
+      children,
+      "aria-labelledby": ariaLabelledBy,
+      style,
+      ...rest
+    },
+    refProp,
   ) {
-    const buttonBarRef = useRef(null);
+    const targetWindow = useWindow();
+    useComponentCssInjection({
+      testId: "salt-carousel-slide",
+      css: carouselSlideCss,
+      window: targetWindow,
+    });
+    const slideRef = useRef<HTMLDivElement>(null);
+    const [index, setIndex] = useState<number | null>(null);
+    const {
+      slidesCount,
+      firstVisibleSlide,
+      registerSlide,
+      unregisterSlide,
+      visibleSlides,
+    } = useCarousel();
 
+    useEffect(() => {
+      if (slideRef.current) {
+        const assignedIndex = registerSlide(slideRef.current);
+        setIndex(assignedIndex);
+      }
+      return () => {
+        if (index !== null) {
+          unregisterSlide(index);
+        }
+      };
+    }, [registerSlide, unregisterSlide]);
```

Comment:
this will not cleanup as index is not in the dependencies ?

Reply:
index would only change when a slide is registered